src/compiler.rs

```diff
@@ -107,11 +110,54 @@ impl<F> CompilationUnit<F> {
     }
 }
 
+impl<F, TraceArgs> From<&astCircuit<F, TraceArgs>> for CompilationUnit<F> {
+    fn from(ast: &astCircuit<F, TraceArgs>) -> Self {
+        CompilationUnit::<F> {
+            annotations: {
+                let mut acc = ast.annotations.clone();
+                for step in ast.step_types.values() {
+                    acc.extend(step.annotations.clone());
+                }
+
+                acc
+            },
+            step_types: ast.step_types.clone(),
+            forward_signals: ast.forward_signals.clone(),
+            shared_signals: ast.shared_signals.clone(),
+            fixed_signals: ast.fixed_signals.clone(),
+            ..Default::default()
+        }
+    }
+}
+
+pub struct CompilerConfig<CM: CellManager, SSB: StepSelectorBuilder> {
+    cell_manager: CM,
+    step_selector_builder: SSB,
+}
```

Comment:
What's the use of `CompilerConfig` as it's literally the same as `Compiler`. I only see it used once in super circuit.

Reply:
It is used here: https://github.com/privacy-scaling-explorations/chiquito/pull/58/files#diff-477cf4d3133e8c73e0a761339f1d361e646943fe2f8f5c332ccdf96df759ed66R31

I think in the end the compiler will not need to contain this fields, they will be removed.

Reply:
Kk. I saw it's used in the link. Just saying that it's the same struct as Compiler basically. Good with eventually removing them! :)